code/modules/reagents/machinery/thermometer.dm

```diff
@@ -47,6 +47,14 @@
 		var/obj/machinery/bunsen_burner/B = target
 		if(B.held_container)
 			target = B.held_container
+	if(istype(target, /obj/machinery/chemheater))
```

Comment:
If someone can give a better way to do this it would be greatly appreciated.

Reply:
A machinery-level proc get_held_container(), check once to see if it's machinery then call that.

You could also do something even more general: remove all the exception cases and instead go to the if statement below this at line 59. In the case that target.reagents = null, locate() a child of reagent_containers in target and set target equal to that. If still null, return.